app/components/agenda/agenda-header/agenda-version-actions.js

```diff
@@ -438,9 +437,10 @@ export default class AgendaAgendaHeaderAgendaVersionActions extends Component {
       await this.reloadAgenda(this.args.currentAgenda);
       await this.reloadAgendaitemsOfAgenda(this.args.currentAgenda);
       await this.reloadMeeting();
-      if (agendaitemsNotOk?.length) {
-        await this.decisionReportGeneration.regenerateDecisionReportsForMeeting.perform(this.args.meeting, true);
-      }
+      // Regenerate ALL beslissingsfiches, so we ensure that the concerns part
+      const newNames = true;
+      const shouldRegenerateConcerns = true;
+      await this.decisionReportGeneration.regenerateDecisionReportsForMeeting.perform(this.args.meeting, newNames, shouldRegenerateConcerns);
```

Comment:
This seems to work fine, but I'm concerned we'll be regenerating too much for large agendas with a few documents added in the last version. This would not only lead to an unnecessary wait/use of server resources, but also would mean the kanselarij needs to re-check all decisions on every approval, even the ones they didn't touch.

I do get why this was the easier route to take, and we discussed it in the standup and I said it would be fine iirc.

Obviously we can't simply look at the agendaitems  with documents without an `originalName`, since we just named them all.

I'm wondering if we can somehow leverage the DocumentNamingJob to filter it down to only the necessary decisions. I'll look into this for a bit.

Reply:
I think I almost have it figured out, will continue tomorrow

Reply:
Made a small adjustment to `document-naming-service` for this, and at least locally, it seems to work now